main.go

```diff
@@ -42,11 +47,24 @@ func main() {
 			if err != nil {
 				return err
 			}
-			select {
-			case <-exit:
-				return nil
-			case err := <-s.Serve():
-				return err
+
+			reload := make(chan os.Signal, 1)
+			signal.Notify(reload, syscall.SIGHUP)
```

Comment:
It would be great to avoid manual steps for reload.

A simple periodic check to trigger reload would avoid manual intervention when a PR changes the backends, and lets CD take care of it.